18_reinforcement_learning.ipynb

```diff
@@ -1849,8 +1849,7 @@
     "    keras.layers.Dense(n_outputs)\n",
     "])\n",
     "\n",
-    "target = keras.models.clone_model(model)\n",
-    "target.set_weights(model.get_weights())"
+    "target = keras.models.clone_model(model)"
```

Comment:
Surprisingly `clone_model()` does **not** copy the weights, it just copies the model architecture. Therefore the `set_weights()` call is actually required.

Here's some proof. Try running this code:

```python
import numpy as np
import tensorflow as tf
from tensorflow import keras

model = keras.models.Sequential([
    keras.layers.Dense(10, activation="relu"),
    keras.layers.Dense(1)
])
model.compile(loss="mse", optimizer="adam")
X_train = tf.random.uniform((100, 5))
y_train = tf.random.uniform((100, 1))
history = model.fit(X_train, y_train)

target = keras.models.clone_model(model)

for w1, w2 in zip(target.weights, model.weights):
  print(np.allclose(w1, w2))
```

Reply:
The reason I deleted this line is because of lines 1930 and 1931: `if episode % 50 == 0: target.set_weights(model.get_weights())`. I thought that because of that, we could do without line 1853.